modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultConstantFoldingOptimizationPhase.java

```diff
@@ -45,6 +46,42 @@
  */
 public class DefaultConstantFoldingOptimizationPhase extends IRExpressionModifyingVisitor {
 
+    private static RuntimeException unaryError(String type, String operation, String constant) {
+        return new IllegalStateException(
+            Strings.format(
+                "constant folding error: unexpected type [%s] for unary operation [%s] on constant [%s]",
+                type,
+                operation,
+                constant
+            )
+        );
+    }
+
+    private static RuntimeException binaryError(String type, String operation, String constant1, String constant2) {
+        return error(type, "binary", operation, constant1, constant2);
+    }
+
+    private static RuntimeException booleanError(String type, String operation, String constant1, String constant2) {
+        return error(type, "boolean", operation, constant1, constant2);
+    }
+
+    private static RuntimeException comparisonError(String type, String operation, String constant1, String constant2) {
+        return error(type, "comparison", operation, constant1, constant2);
+    }
+
+    private static RuntimeException error(String type, String opType, String operation, String constant1, String constant2) {
```

Comment:
Why return RuntimeException instead of ISE?

Reply:
Left over from a previous version of the method